widget/nsIWidget.h

```diff
@@ -2087,8 +2087,18 @@ class nsIWidget : public nsISupports {
     virtual void SetDrawMouse(bool aDrawMouse)
     { mDrawMouse = aDrawMouse; }
 
+    virtual void SetDrawVirtualMouse(bool aDrawVirtualMouse)
+    { mDrawVirtualMouse = aDrawVirtualMouse; }
+
     virtual bool GetDrawMouse()
-    { return mDrawMouse; }
+    { return mDrawMouse || mDrawVirtualMouse; }
+
+    /**
+     * For remote control draw mouse cursor in nsContentUtils.
+     * widget/gonk/nsWindow.cpp has real implementation.
+     * Otherwise, use empty implementation.
+     */
+    virtual void ForceComposition() { };
```

Comment:
I put ForceComposition in nsIWidget because nsContentUtils can access it, not nsWindow.
Use #ifdef MOZ_WIDGET _GONK might be ok, but caller also need to add #ifdef, otherwise other platform like b2g-desktop may get error. That's discuss a better solution for this.